### ISSUE TITLE
Checking for broken visuals

### ISSUE THREAD

**Issue by flavio-meneses** [pbi-tools/pbi-tools#261]
Hi Mathias, question if I may. I'm trying to look for broken visuals in a report automatically, as per image below.

![image](https://user-images.githubusercontent.com/59139011/230730014-3c234490-f06e-47ff-9644-ddadc00c20ba.png)

I've spotted a pattern in which broken visuals don't have a "query.json" or "dataTransforms.json", which allows checking for broken visuals easily. 

However, I've just noticed an inconsistent behaviour in which some visuals that are OK also don't have both .jsons:

<img width="512" alt="image" src="https://user-images.githubusercontent.com/59139011/230730133-b8bf9218-ad8d-4c0a-9f6e-188625f2966a.png">
   
Those .json files seem to be created at some random point later, when I run pbi-tools again: 
 
<img width="472" alt="image" src="https://user-images.githubusercontent.com/59139011/230730186-9f6d670b-fc5e-48bc-a2fc-460d410594f2.png">

Are you aware of what's triggering these files to be generated? Or an easier way to look for broken visuals? Currently I haven't found a reliable way to flag this.

Thanks!
Flavio

**Comment by mthierba** [comment]
Hi Flavio,

That's a bit of a red herring, I'm afraid. Those two files, `query.json` and `dataTransforms.json`, are entirely optional and only contain cached data. Their presence or absence doesn't indicate anything about the validity of a visual.
Mathias

**Comment by flavio-meneses** [comment]
Thanks Mathias, I'll have to drop this check then until a more robust way to do this is found. 

If you're aware of an alternative please let me know, it would be extremely powerful to have this assurance on large reports that are laborious to check!

Cheers,
Flavio

**Comment by BVMather** [comment]
Hi Flavio, one way - albeit initially manual. If you use Performance Analyser, Start Recording, Refresh Visuals and then Export then you can capture the DAX queries for all visuals on a page to a json file.  You can then write a test that iterates through each json file and executes each DAX query and flags if there is an error.

**Comment by flavio-meneses** [comment]
> Hi Flavio, one way - albeit initially manual. If you use Performance Analyser, Start Recording, Refresh Visuals and then Export then you can capture the DAX queries for all visuals on a page to a json file. You can then write a test that iterates through each json file and executes each DAX query and flags if there is an error.

Hi Brian, that would be a good avenue to explore actually, as the broken visual doesn't seem to generate a query (makes sense). Too bad the REST API doesn't expose any of this functionality, if you find any way to run the analyser on big reports automatically let me know! Cheers.

**Comment by BVMather** [comment]
Yes, if you can capture the queries when they are working - what you'll find is that the query execution will fail.  Will let you know if we come up with anything
